### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by noelfoka**
## Bookstore: Use Redux in React components.

In this branch, I:

- [ ] Wrap `<App>` with the `<Provider>` component from `react-redux`.
- [ ] Pass the configured `store` into the `<Provider>`.
- [ ] Create an `initialState` variable for your `booksSlice`.
- [ ] Display the `books`, received from the slice, in a reusable component.
- [ ] Dispatch actions using `useDispatch`.
- [ ] Add a `<Button>` component, which includes:
    - An event handler that adds a book to the books array.
    - An event handler that removes a book from the books array

### CODE REVIEW

File: src/components/BookForm.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        value={title}
+        onChange={(e) => setTitle(e.target.value)}
+      />
+      <input
+        type="text"
+        placeholder="Author"
+        value={author}
+        onChange={(e) => setAuthor(e.target.value)}
+      />
+      <button type="submit">ADD BOOK</button>
```
Comment (FrederickMih) [inline src/components/BookForm.jsx]: - [ ] Hi @noelfoka, the event handler(onClick) to add a book to the book list should be implemented on this `<button></button>` as you can see on the screenshot below not on the form itself. Let's kindly fix :wrench: it

Screenshot from 2023-08-23 05-20-18

File: src/components/Book.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from './Button';
+
+const Book = ({ book, onDelete }) => (
+  <div>
+    <h3>{book.title}</h3>
+    <p>{book.author}</p>
+    <Button onClick={onDelete}>Delete</Button>
```
Comment (FrederickMih) [inline src/components/Book.jsx]: - [ ] Let's kindly refer to the comment above :arrow_up: This button should include an event handler to remove a book from the books array with the id attribute and the removeBook action should be dispatch

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @noelfoka, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Books are accessed inside the BookList component using useSelector as requested :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookForm.jsx (hunk 1 above)
Comment (Johnadibe) [reply src/components/BookForm.jsx]: > - [ ] As requested by the code reviewer, kindly add the event handler(onClick) to add a book to the book list on the button, not the form itself.👍

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️
### Hi @noelfoka🙋‍♂️, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
- Implemented some changes requested by the previous reviewer✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @noelfoka, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Professional `README` template ✅ 
- Linters are green ✅ 
- Correct `Gitflow` ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**